lab-cortney/controller/auth-controller.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+// auth-controller takes a req.body containing a new username and password,
+// utilizes the User model to ensure unique username and hash,
+// then sends user a token of... authenticity? for what?
+
+const debug = require('debug')('authlab:auth-controller');
+
+const User = require('../model/user');
+
+exports.signup = function(reqBody){
+  debug('signup');
+  return new Promise((resolve, reject) => {
+    var password = reqBody.password;
+    delete reqBody.password; // so... we're deleting the user's pw right off the bat?
```

Comment:
I talked to Duncan about this a few days ago. Right now, we are not learning about security, so you are temporarily storing the password in your variable (line 14), and deleting it on (line 15), and that is ok. Reason why we delete it is that, in real life, you do not want to save non-hashed password anywhere (even in variables), so we pass them along to where we can hash it, then later store it into your db.